### DISCUSSION TITLE
Emby as Music Provider

### DISCUSSION THREAD

**Discussion by K3str3l** [music-assistant/support#583] [Feature requests and ideas]
Hello, I would like to be able to use Emby Server as a music provider.

**Comment by rfporter** [comment]
I tried adding Emby as a Jellyfin provider. It didn't work.
Had to spin up a Jellyfin server just for music.

**Reply by OzGav** [reply]
You would be better served just allowing MA to access your music files directly

**Reply by vincegre** [reply]
> You would be better served just allowing MA to access your music files directly

Not possible when server is remote and also doesn't allow to share playlist so pointless sorry to say :(

**Reply by OzGav** [reply]
I don't know why you thumbs down the other users post. It was a useful contribution for others in the future. 

And my post is not **pointless**.  Just because you might have some reason to not use local files does not mean that it isn't preferable for those that don't have your constraints. I will assume you made a poor choice of words.

**Reply by rfporter** [reply]
Thanks for both of your comments. It would be great to have a best practices section somewhere. 

I wanted to use emby in the hope that I could reuse playlists and favorites. Its annoying to have to redo them for each new server and player that I'm trying out. 

Giving MA access to my local files is an option since I'm using a VPN for remote access. 

If Symfonium supported MA as a provider (or if MA had a mobile app that supported offline cache), then I think that MA with direct file access would become the preferred music provider.

**Reply by Jc2k** [reply]
Hi. I maintain the Jellyfin provider. It works quite well for people running Jellyfin.

There was some interest in Emby via the Jellyfin provider - see https://github.com/music-assistant/server/pull/1325. So it did work at one point. We are really hitting the places where those projects have diverged too much now and might be it will need its own provider. Or it could be that you have an unusual file in your emby library that breaks some assumption we have. It's not clear /why/ it didn't work for you.

If you have a folder full of music that is well tagged (using something like Picard) and you don't need any of the apps in the Jellyfin ecosystem (e.g. Finamp) then @OzGav is right, spinning up an entire Jellyfin just for music is probably not the best use of your resources. Personally, I run multiple MA's in different locations backed by the same "production" music library in Jellyfin (e.g. home, office over vpn, dev environment) whilst still using apps like Finamp etc as well. And i'd be running it anyway for other media. So it's unlikely i'd ever switch to files.

**Reply by rfporter** [reply]
I'm just going to throw this out here: https://pypi.org/project/embyclient/
Also: https://emby.media/community/index.php?/blogs/entry/544-presenting-the-new-emby-sdk/

**Reply by spaceboy76** [reply]
also looking to use emby as a source. my files are local but my file structure (which works fine with emby) unfortunately does not work fully with Music Assistant file system source. i will set up and use Jellyfin as an interim measure

**Reply by OnkelBuTscH** [reply]
Yes, I would also be very happy to be able to integrate emby into music assitant.

**Reply by RandomLegend** [reply]
I just went full into music assistant + squeezeboxes (and squeezelite clients) and i love it! But jellyfin is really lacking when it comes to music features; The biggest one missing - Dynamic / Smart Playlists based on genre (they have it for movies / tv but not Music?????) and Emby is widely viewed as superior for music.

I'd love to have a dedicated Emby Music Provider :) (Navidrome would also be very cool instead i guess)

**Reply by rfporter** [reply]
Navidrome works with the subsonic music provider. 
I went full (self hosted) circle. Emby -> Jellyfin -> Navidrome -> Gonic -> LMS -> Back to Emby. 
Emby is the best at managing my music library. The others are good, but not as polished as Emby. 

Navidrome had issues with multi-artist albums / tracks. 

LMS works great with Musicbrainz-tagged collections but doesnt work with subsonic provider in Music Assistant.

For now I am just using local filesystem as music provider in music assistant as I wait and hope for an Emby provider to be developed.

**Reply by mptpro** [reply]
I also vote for Emby integration.  Among Jellyfin, Navidrome, Plex and Emby I found that Emby works the best for managing music (for me at least). I've have tried using the JellyFin provider as another poster above did, but I couldn't get it to work with Emby.

thanks.

**Reply by Virtualchronos** [reply]
I also need emby integration, it's the best player !

**Reply by OzGav** [reply]
I’m more interested to know why you want to use Emby vs MA directly? You can use Emby as well but why not just let MA access your local media directly?

**Reply by mptpro** [reply]
> I’m more interested to know why you want to use Emby vs MA directly? You can use Emby as well but why not just let MA access your local media directly?

Primarily for the many playlists that I already have in Emby. But for now I _have_ been accessing my local media directly via MA. 

I do have question... if Emby ever gets added, like Jellyfin, do playlists and favorites get syncronized between the two?

**Reply by vincegre** [reply]
> I’m more interested to know why you want to use Emby vs MA directly? You can use Emby as well but why not just let MA access your local media directly?

Because often media server is not local so it's not an option and also you don't have access at playlists in that case !

**Reply by OzGav** [reply]
@mptpro Not that there has been any work done on this at all but if whatever API that Emby uses supports sync of playlists and favourites then yes I expect that would be provided just like every other provider. 

@vincegre If you collocate the MA server with your music then you can playback anywhere these days.

**Reply by vincegre** [reply]
> @vincegre If you collocate the MA server with your music then you can playback anywhere these days.

Impossible as I don't plan to install an HA instance at the rack in datacenter where my emby server is running ;) and same issue with people that use an account on an emby of third party ;)

**Reply by RandomLegend** [reply]
> I’m more interested to know why you want to use Emby vs MA directly? You can use Emby as well but why not just let MA access your local media directly?

As @rfporter i also use Symfonium as an app; Primarily because of its Android Auto support. So i need a music backend that lets me sync my playlists to Symfonium, i also need a backend that supports .m3u playlist files i just drop into a folder because my wife has a MP3 Player that she loves to use and that can handle .m3u 

So to conclude, i have several devices / apps that need to be able to sync with  my Music Backend and/or my backend has to be able to just spit out the playlist files i made for my wifes MP3 player.

MA doesn't allow me to either sync to Symfonium, nor gives out the .m3u playlist files.

**Reply by mptpro** [reply]
> MA doesn't allow me to either sync to Symfonium, nor gives out the .m3u playlist files.

I actually have this issue as well, just didn't mention it above. Emby is great for mobile streaming on Android, both using the Emby app and also that it works with apps such as Symfonium,

**Reply by spaceboy76** [reply]
> I’m more interested to know why you want to use Emby vs MA directly? You can use Emby as well but why not just let MA access your local media directly?

its playlists for me too

**Reply by K3str3l** [reply]
I can use Emby for audio / images / video on my TV & phones. Rather than duplicate effort (playlist as an example) would like to be able to have one "source".

**Reply by neurocis** [reply]
+1 as well for playlists, cross platform integration ( I use Symfonium as well as Emby client on iOS ) also our whole family utilizes favorites stored by Emby.

**Reply by mattiasghodsian** [reply]
+1 for emby support

**Reply by FrancisGalton** [reply]
+1

**Reply by No-Idea-76** [reply]
+1

**Reply by maisun** [reply]
+1

**Comment by OzGav** [comment]
Unlucky for some but not for you guys. This will be available in 2.8.0 beta 13

**Reply by maisun** [reply]
Wow great news! Thanks a lot!!!

**Reply by vincegre** [reply]
Is the beta already available ? Would be happy to give feedback 😉

**Reply by rfporter** [reply]
Thanks @OzGav and @hatharry and the music assistant team! I'm looking forward spinning up music assistant again to try this out.

**Comment by hmsGhostship** [comment]
I've tried the EMBY Provider with MusicAssistant 2.8.0b20.
I've noticed that the track order in an album is always alphabetical and can't be changed in the sort order settings.
I assume this is a bug; where can I report it?

**Reply by OzGav** [reply]
@hatharry

**Comment by hmsGhostship** [comment]
I'm not really familiar with creating bug reports in GitHub. I'd rather stick with Jellyfin; that works. ;)

**Comment by hatharry** [comment]
https://github.com/music-assistant/server/pull/3400

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
